src/features/processPicker.ts

```diff
@@ -64,28 +140,47 @@ abstract class DotNetAttachItemsProvider implements AttachItemsProvider {
         return this.getInternalProcessEntries().then(processEntries => {
-            // localeCompare is significantly slower than < and > (2000 ms vs 80 ms for 10,000 elements)
-            // We can change to localeCompare if this becomes an issue
-            let dotnetProcessName = (os.platform() === 'win32') ? 'dotnet.exe' : 'dotnet';
-            processEntries = processEntries.sort((a, b) => {
-                if (a.name.toLowerCase() === dotnetProcessName && b.name.toLowerCase() === dotnetProcessName) {
-                    return a.commandLine.toLowerCase() < b.commandLine.toLowerCase() ? -1 : 1;
-                } else if (a.name.toLowerCase() === dotnetProcessName) {
-                    return -1;
-                } else if (b.name.toLowerCase() === dotnetProcessName) {
-                    return 1;
-                } else {
-                    return a.name.toLowerCase() < b.name.toLowerCase() ? -1 : 1
-                }
-            });
-            
-            let attachItems = processEntries.map(p => p.toAttachItem());
-            return attachItems;
+            return sortProcessEntries(processEntries, os.platform())
         });
     }
 }
 
+function sortProcessEntries(processEntries : Process[], osPlatform : string) : AttachItem[] {
+    // localeCompare is significantly slower than < and > (2000 ms vs 80 ms for 10,000 elements)
+    // We can change to localeCompare if this becomes an issue
+    let dotnetProcessName = (osPlatform === 'win32') ? 'dotnet.exe' : 'dotnet';
+    processEntries = processEntries.sort((a, b) => {
+        if (a.name.toLowerCase() === dotnetProcessName && b.name.toLowerCase() === dotnetProcessName) {
+            return a.commandLine.toLowerCase() < b.commandLine.toLowerCase() ? -1 : 1;
+        } else if (a.name.toLowerCase() === dotnetProcessName) {
+            return -1;
+        } else if (b.name.toLowerCase() === dotnetProcessName) {
+            return 1;
+        } else {
+            return a.name.toLowerCase() < b.name.toLowerCase() ? -1 : 1
+        }
+    });
+
+    let attachItems = processEntries.map(p => p.toAttachItem());
+    return attachItems;
+}
+
 export class PsAttachItemsProvider extends DotNetAttachItemsProvider {
+    protected getInternalProcessEntries(): Promise<Process[]> {
+        const commColumnTitle = Array(PsOutputParser.secondColumnCharacters).join("a");
+        // the BSD version of ps uses '-c' to have 'comm' only output the executable name and not
+        // the full path. The Linux version of ps has 'comm' to only display the name of the executable
+        // Note that comm on Linux systems is truncated to 16 characters:
+        // https://bugzilla.redhat.com/show_bug.cgi?id=429565
+        // Since 'args' contains the full path to the executable, even if truncated, searching will work as desired.
+        const psCommand = `ps -axww -o pid=,comm=${commColumnTitle},args=` + (os.platform() === 'darwin' ? ' -c' : '');
```

Comment:
as.platform() is going to tell you the host platform, you want the target platform